dartagnan/src/main/antlr4/SpirvLexer.g4

```diff
@@ -1691,3 +1692,51 @@ LiteralExtInstInteger : [A-Za-z] [A-Za-z0-9_]*;
 ModeExt_LineComment : ';' ~[\n]* -> skip;
 ModeExt_Whitespace : [ \t\r]+ -> skip;
 ModeExt_NewLine : [\n]+ -> popMode, skip;
+
+mode ModeHeader;
+
+ModeHeader_Input    : 'Input';
+ModeHeader_Output   : 'Output';
+ModeHeader_Config   : 'Config';
+
+ModeHeader_Op       : 'Op' -> type(Op);
+ModeHeader_Id       : '%' [A-Za-z0-9_]+ -> type(Id);
+
+AssertionExists     : 'exists';
+AssertionForall     : 'forall';
+AssertionFilter     : 'filter';
```

Comment:
I think it depends on whether we plan to explore DR with SPIRV parser? @hernanponcedeleon

Reply:
The filter thing does not really depend on checking data races. It is just a way of saying the tool "just explore the subset of the executions that satisfy this property".

For the Vulkan litmus tests, we just used it because the original tests has the filter. I don't think we will use this at all for spirv programs (we don't have it either for C programs).

Reply:
For `C` programs we have `__VERIFIER_assume` though. I believe such a feature is sufficient for `SPIRV` as well, i.e., you can mimic the strange litmus filters by placing `__VERIFIER_assume` accordingly.

Reply:
Okay, I'll just drop it for now if we don't need it urgently, we can add this `__VERIFIER_assume` afterward when we need filter.